internal/charmstore/store.go

```diff
@@ -1425,6 +1425,102 @@ func (store *Store) Search(sp SearchParams) (SearchResult, error) {
 	return result, nil
 }
 
+var listFilters = map[string]string{
+	"name":        "name",
+	"owner":       "user",
+	"series":      "serties",
+	"type":        "type",
+	"promulgated": "promulgated-revision",
+}
+
+func prepareList(sp SearchParams) (filters map[string]interface{}, sort []string, err error) {
```

Comment:
This also needs to check that some of the search-only parameters aren't set.
For example, if any of the Text, AutoComplete, Limit or Skip fields are set
in sp, we should return an error.